text/0000-temp-components-lib.md

```diff
@@ -0,0 +1,610 @@
+
+1. [Custom hooks](#custom-hooks)
+2. [Helper functions](#helper-functions)
+
+_:bulb: In addition to React 18, all components should be natively compatible for use
+within React Server Components._
+
+### Simple Components
+
+Simple components should mimick the vanilla JSX design React provides with the Headless-styles
+props extending it (i.e. not manipluataing React props or creating "component
+does all" solutions).
+
+For example, a Button should be built something like this:
+
+```typescript
+import { getButtonProps } from '@pluralsight/headless-styles'
+import type { ButtonOptions } from '@pluralsight/headless-styles/types'
+import { type HTMLAttributes, type PropsWithChildren } from 'react'
+
+interface ButtonProps extends ButtonOptions, HTMLAttributes<HTMLButtonElement> {}
+
+export function Button(props: PropsWithChildren<ButtonProps>) {
+  const {
+    className,
+    disabled,
+    icon,
+    sentiment,
+    size,
+    usage,
+    ...nativeBtnProps
+  } = props
+  const pandoProps = getButtonProps({
+    className,
+    disabled,
+    sentiment,
+    size,
+    usage,
+    icon
+  })
+
+  return (
+    <button {...pandoProps.button} {...nativeProps} />
+  )
+}
+```
+
+By nature, this example uses the native React JSX design while extending the
+Headless-styles props on top of it so it can be used in a way that compliments
+and teaches users about React & Javascript due to not manipluating any props/React APIs.
+
+We are purposefully not using `forwardedRef` to prevent complexity and possible
+performance implications. This idea might need to change, but we should wait until
+we have enough evidence the example below does not work in most use cases.
+
+This is an example of how the Button above can be used:
+
+```typescript
+<span ref={btnRef}>
+  <Button type="button" onClick={handleClick}>
+    <ShareIcon hidden />
+    Share
+  </Button>
+</span>
+```
+
+### Complex Components
+
+For more complex components, we should be able to lean into the same design above
+by holding true to our standard of keeping the library designed in a composable way.
+Likewise, most complex components should include a Provider to manage state and/or
+data when needed.
+
+For example, a Tabs family would be composable presentational components that
+include a Provider for state management only, not the data going into them.
+
+This keeps Pando's responsibility clean meaning that Pando should only be
+managing UI states, not the data being used within its components.
+
+_:warning: This would require refactor from the original [Tabs Hooks](https://design.pluralsight.com/docs/reference/react-hooks/useTabs) which are mostly composable, but would need to be
+pushed even further for the ideal design. Likewise, they would need to be moved
+into the new react library for colocation._
+
+An example of the Tabs family in practice, should look something like this:
+
+```jsx
+<TabsProvider>
+  <Tabs>
+    <TabList>
+      <Tab>Tab 1</Tab>
```

Comment:
How could a user modify a click handler, to add a analytics tracker for example?  Create a custom component using the provided useTabProps hook?

Reply:
@alexzywiak great question, since the new library would be composable and just extend the native React element we use, you should be able to put click handlers on any element that React naturally allows. No need for custom components or ejecting necessary.